src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx

```diff
@@ -119,8 +121,13 @@ export const createQueryAssistExtension = (
         };
       }
     },
-    isEnabled$: () =>
-      getAvailableLanguages$(http, data).pipe(map((languages) => languages.length > 0)),
+    isEnabled$: (dependencies) => {
+      const query = dependencies.query;
+      if (!queryAssistantSupportedDatasetTypes.includes(query.dataset?.type || '')) {
+        return new BehaviorSubject(false);
```

Comment:
Nit: what about simply do:
```suggestion
        return of(false);
```

Reply:
Sure, updated.